docs/architecture/clients/services/implementation.md

```diff
@@ -0,0 +1,200 @@
+
+For each of these scenarios, we will examine how the classes above work to persist and propagate
+changes across clients.
+
+### Server Update: Updated in the current client
+
+```kroki type=plantuml
+@startuml
+abstract class Internal[Domain]Service #aliceblue {
+  + upsert(item: [Domain])
+  + replace(items: [Domain][])
+}
+class [Domain]Service #aliceblue {
+  # subject : BehaviorSubject<[Domain][]>
+  + observable: Observable<[Domain][]>
+  - updateObservables()
+}
+class [Domain]ApiService #aliceblue {
+  + put[Domain]()
+}
+together {
+class "UI Component"
+class "Other UI Component"
+}
+class ApiService {
+  + send()
+}
+
+"UI Component" --> "[Domain]ApiService"
+"[Domain]ApiService" --> ApiService
+"[Domain]ApiService" --> "Internal[Domain]Service"
+"Internal[Domain]Service" --> "[Domain]Service"
+"[Domain]Service" ..> "Other UI Component" : Observable next()
+@enduml
+```
+
+In the case that the domain is updated in the client, the change will begin with the user modifying
+data in a UI component. The UI component has a dependency on `[Domain]ApiService`, which is
+responsible for transmitting that change to two places:
+
+1. To the server, to persist the change and also notify other clients, and
+2. To the `BehaviorSubject` for that domain that is storing the state of that domain for the current
+   client
+
+You can see those responsibilities above, where `ApiService` updates the server and
+`Internal[Domain]Service` updates the `BehaviorSubject`.
+
+:::note
+
+In our current implementation, `Internal[Domain]Service` is implemented by `[Domain]Service`.
+However, that implementation is abstracted from the `[Domain]ApiService` so that there is not a
+direct dependency. This will allow for independent implementations in the future.
+
+:::
+
+### Cache-Only Update: Updated in a different client
+
+```kroki type=plantuml
+@startuml
+abstract class Internal[Domain]Service #aliceblue {
+  + upsert(item: [Domain])
+  + replace(items: [Domain][])
+}
+class [Domain]Service #aliceblue {
+  # subject : BehaviorSubject<[Domain][]>
+  + observable: Observable<[Domain][]>
+  - updateObservables()
+}
+together {
+    class "UI Component"
+    class SyncService {
+    + fullSync()
+    - sync[Domain]()
+    }
+}
+
+SyncService --> "Internal[Domain]Service"
+"Internal[Domain]Service" --> "[Domain]Service"
+"[Domain]Service" ..> "UI Component" : Observable next()
+
+@enduml
+```
+
+For domain updates from another client, the current client receives those messages through the
+`SyncService`. The `SyncService` takes a dependency on the `Internal[Domain]Service`, as it is
+responsible for updating only the internal cache. The `SyncService` updates the domain through this
+service to propagate the changes in the current client.
+
+The `Internal[Domain]Service` uses `BehaviorSubject`s in the `[Domain]Service` for that caching
+mechanism, but that implementation is abstracted from the `SyncService` and any other consumers of
+`Internal[Domain]Service`.
```

Comment:
Not all domains cache, this is an implementation detail of the domain. You point that out, but maybe change `uses` to `may use` to indicate it's not a guarantee.

Reply:
Done. Good idea.